### DISCUSSION TITLE
v1.0.29

### DISCUSSION THREAD

**Discussion by hsakoh** [hsakoh/switchbot-mqtt#54] [Announcements]
**Full Changelog**: https://github.com/hsakoh/switchbot-mqtt/compare/v1.0.28...v1.0.29
- Support for the following new documented device has been added:
    - W3002500 Robot Vacuum K10+ Pro Combo
    - W3902300 Evaporative Humidifier
    - W3902310 Evaporative Humidifier (Auto-refill)
    - W5302300 Air Purifier PM2.5
    - W5302310 Air Purifier Table PM2.5	
    - W5302300 Air Purifier VOC
    - W5302310 Air Purifier Table VOC
    - W5000000 Roller Shade
    - W5502310 Relay Switch 1PM
    - W5502300 Relay Switch 1

<hr /><em>This discussion was created from the release <a href='https://github.com/hsakoh/switchbot-mqtt/releases/tag/v1.0.29'>v1.0.29</a>.</em>

**Comment by hsakoh** [comment]
Support for multiple devices has been added.
Although implementation is done according to the documentation, it’s common for the actual API responses and Webhook payloads to differ from what’s documented.
If a new device does not function properly on HomeAssistant, please provide the following information:
Specific API responses, Webhook payloads, command API requests and responses, etc. You can obtain these from the logs by setting the log level to "trace" in the add-on settings.
Feedback on devices that are functioning correctly is also welcome.

**Comment by danielsza** [comment]
initial testing for W3902310 Evaporative Humidifier (Auto-refill);
all functions appear to work correctly.

I will report back if any issues occur.

Thank you for quick respone and getting support added so quickly.

**Comment by Glup3** [comment]
Just received my K10+ Pro, installed the add-on via docker container and it works! Thank you so much for this awesome tech!
